shopfloor_workstation/services/workstation.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2021 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo.addons.base_rest.components.service import to_int
+from odoo.addons.component.core import Component
+
+
+class ShopfloorWorkstation(Component):
+    """
+    Workstation service for the mobile application.
+
+    This allows to scan a workstation bar code on the floor, which
+    will set a default profile on the mobile app and offer the option
+    to change some configuration of the connected user.
+
+    """
+
+    _inherit = "base.shopfloor.service"
+    _name = "shopfloor.workstation"
+    _usage = "workstation"
+    _expose_model = "shopfloor.workstation"
+    _description = __doc__
+
+    def setdefault(self, barcode):
+        """Endpoint that receives a scanned barcode."""
+        ws = self.env["shopfloor.workstation"].search([("barcode", "=", barcode)])
+        if self._set_workstation_as_default(ws):
+            message = {
+                "message_type": "info",
+                "body": "Default workstation set to {}".format(ws.name),
+            }
+        else:
+            message = {
+                "message_type": "error",
+                "body": "Workstation not found",
+            }
+        return self._response(
+            message=message, data={"size": len(ws), "records": self._to_json(ws)}
+        )
+
+    def _set_workstation_as_default(self, workstation):
+        """Apply changes to backend user configuration."""
+        if not workstation:
+            return False
+        if workstation.printing_printer_id:
+            # TODO : should the default action be checked ?
+            self.env.user.printing_printer_id = workstation.printing_printer_id
```

Comment:
Does the user always has permission to write on this field?

Reply:
Yes he should, it is taken care of there https://github.com/OCA/report-print-send/blob/2160eedf36248d5b7660c5e41a4f03408271668c/base_report_to_printer/models/res_users.py#L28